python/pacemaker/_cts/network.py

```diff
@@ -0,0 +1,34 @@
+""" Network related utilities for CTS """
+
+__all__ = ["next_ip"]
+__copyright__ = "Copyright 2000-2023 the Pacemaker project contributors"
+__license__ = "GNU General Public License version 2 or later (GPLv2+) WITHOUT ANY WARRANTY"
+
+def next_ip(env):
+    """ Return the next available IP address """
+
+    ip = env["IPBase"]
+
+    new_ip = None
+
+    # Split the existing IP address up into a tuple of:
+    # (everything except the last part of the addr, the separator, the last part of the addr).
+    # For instance, "192.168.1.2" becomes ("192.168.1", ".", "2").  Then,
+    # increment the last part of the address and paste everything back
+    # together.
+    if ":" in ip:
+        # This is an IPv6 address
+        fields = ip.rpartition(":")
+        new_ip = hex(int(fields[2], 16) + 1)
+
+    else:
+        # This is an IPv4 address
+        fields = ip.rpartition(".")
+        new_ip = int(fields[2]) + 1
+
+    if new_ip > 255:
+        raise ValueError("No more available IP addresses")
```

Comment:
The last component of an IPv6 address can be up to 0xffff (unless it's a IPV4-mapped IPv6 address, which can be detected by the final component containing '.' and then that treated like IPv4).

IPv6 addresses usually are based on interface IDs rather than assigned in ascending order, but I suppose the user can assign specific IPs to be used.

Reply:
Updated to change the check to be IPvX specific.  I'm not actually sure any of this stuff has been tested with IPv6 recently.  I certainly don't have that set up here.  I wonder if anyone does.

Reply:
This is the perfect function for a unit test.